app/controllers/search_controller.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+class SearchController < ApplicationController
+  VALID_IP_REGEX = /^((25[0-5]|(2[0-4]|1\d|[1-9]|)\d)\.?\b){4}$/i
+  def search
+    @query = params[:q] || ''
```

Comment:
(non blocking comment) What do we want to do if the query is empty? Should we return all results or nothing?
If we don't want to return anything, I'm nitpicking, but we can return early to skip the SQL queries

Reply:
```suggestion
    @query = params[:q] || ''
    return unless @query != ''
```

Reply:
Or simpler, `return if @query == ''`

Reply:
```suggestion
    @query = params[:q] || ''
    return if @query == ''
```

Reply:
Sorry to think slowly, but I wonder if something like this would be even better ^^'
```rb
@query = params[:q]
return if @query.blank?
```
(maybe check the difference between `.empty?` and `.blank?` to be sure it's the right one to use)

Reply:
The first test is only passing with `.blank?` used. This is because the first test make a request with no parameter `q`, thus `@query` is evaluated to `nil` and `.empty?` method is undefined for `nil` object.

The `return` work as intended : the SQL query is skipped for empty query.